Kronos-Shifts-Connector/Microsoft.Teams.Shifts.Integration/Microsoft.Teams.Shifts.Integration.Common/Providers/GraphUtility.cs

```diff
@@ -155,32 +148,56 @@ public class GraphUtility : IGraphUtility
 
+            List<ShiftTeams> shiftTeams = new List<ShiftTeams>();
+            var hasMoreTeams = false;
+
             this.telemetryClient.TrackTrace(BusinessLogicResource.FetchShiftTeamDetailsAsync, fetchShiftUserDetailsProps);
             var hcfClient = this.httpClientFactory.CreateClient("GraphBetaAPI");
             hcfClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", accessToken);
             hcfClient.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
-            using (var httpRequestMessage = new HttpRequestMessage(HttpMethod.Get, "groups")
-            {
-            })
+            var requestUri = "groups";
+            do
             {
-                var response = await hcfClient.SendAsync(httpRequestMessage).ConfigureAwait(false);
-                if (response.IsSuccessStatusCode)
+                using (var httpRequestMessage = new HttpRequestMessage(HttpMethod.Get, requestUri))
                 {
-                    var responseContent = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
-                    return responseContent;
-                }
-                else
-                {
-                    var failedResponseContent = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
-                    var failedResponseProps = new Dictionary<string, string>()
+                    var response = await hcfClient.SendAsync(httpRequestMessage).ConfigureAwait(false);
+                    if (response.IsSuccessStatusCode)
                     {
-                        { "FailedResponse", failedResponseContent },
-                    };
+                        var responseContent = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
+                        var allResponse = JsonConvert.DeserializeObject<AllShiftsTeam>(responseContent);
+                        shiftTeams.AddRange(allResponse.ShiftTeams);
 
-                    this.telemetryClient.TrackTrace(BusinessLogicResource.FetchShiftTeamDetailsAsync, failedResponseProps);
-                    return string.Empty;
+                        // If Shifts has more Teams. Typically graph API has 100 teams in one batch.
+                        // Using nextlink, teams from next batch is fetched.
+                        if (allResponse.NextLink != null)
+                        {
+                            requestUri = allResponse.NextLink.ToString();
+                            hasMoreTeams = true;
+                        }
+
+                        // nextlink is null when there are no batch of teams to be fetched.
+                        else
+                        {
+                            hasMoreTeams = false;
+                        }
+                    }
+                    else
+                    {
+                        var failedResponseContent = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
```

Comment:
You could be stuck in an infinite loop.
Throw on failure.

Reply:
@aosolis , Updated the flag= false  in case of failure case as well.